tensorflow/contrib/image/python/ops/image_ops.py

```diff
@@ -508,6 +511,82 @@ def has_zero():
     else:
       return components
 
+def recursive_fill(mask, threshold, total_dims, max_iter=None):
+  """Recursively iterates until the provided 0-1s 2D or 3D mask is filled.
+  If max_iter is provided, it will iterate for a number of times.
+  If total_dims is 2, threshold is recommended to be 4.
+  If total_dims is 3, threshold is recommended to be 13.
+
+  Args:
+    mask: A 2D or 3D `Tensor` of 0s and 1s representing the mask to be filled.
+    threshold: A threshold indicating sensitive the hole detector will be.
+    total_dims: Number of dimensions of the mask. Only 2 or 3 allowed.
+    max_iter: Number of maximum iterations performed to fill the mask. If
+      max_iter=1 it will be the same as performing "dilation".
+
+  Returns:
+    The mask filled.
+
+  Example:
+
+  ```python
+  init_array = (np.random.random((10,10,10))>0.4)*1
+  inp = tf.placeholder(tf.float32,name="my_input")
+  res_fill = recursive_mask_fill(inp,13,3)
+
+  sess = tf.Session()
+  filled_mask = sess.run(res_fill,feed_dict={inp:init_array})
+  ```
+
+  """
+  if not total_dims in [2, 3]:
+    raise Exception("total_dims is expected to be 2 or 3")
+
+  with ops.name_scope("recursive_fill"):
+    mask = ops.convert_to_tensor(mask, name="mask")
+    orig_dim = array_ops.shape(mask)
+    filter_cube_size = math_ops.pow(3, total_dims)
+
+    filt = variables.Variable(array_ops.ones(filter_cube_size),
+                              validate_shape=False)
+    center_index = math_ops.cast(filter_cube_size/2, dtypes.int32)
+    filt = state_ops.scatter_update(filt, center_index, -99)
+    filt = array_ops.reshape(filt, [3 for _ in range(total_dims)]+[1, 1])
+
+    mask = array_ops.expand_dims(mask, 0)
+    mask = array_ops.expand_dims(mask, -1)
+    new_voxels = array_ops.ones_like(mask)
+
+    # Note: the "y" argument needs to be given because as while_loop returns
+    # two elements, it requires two inputs as well.
```

Comment:
What happens if you remove the second argument, and `loop_body` and `loop_cond` just reference `new_voxels` from the outer scope?